scripts/imgtool/image.py

```diff
@@ -442,6 +442,10 @@ def create(self, key, public_key_format, enckey, dependencies=None,
             else:
                 sig = key.sign_digest(digest)
             tlv.add(key.sig_tlv(), sig)
+        elif fixed_sig is not None and key is None:
```

Comment:
according to bootutil_img_validate() a TLV with public-key/public-key-has is aslo required.